### PR TITLE
Lua fixes

### PR DESCRIPTION
**Pull request by Flamefire**
Some lua fixes like the victory dialog and UTF-8 BOM in lua scripts and some related work as UB due to specializing  `std` traits.

Also fixes the version in the ingame settings

### CODE REVIEW

**Comment by Flamefire** [comment]
@Flow86 Done. There still is this error on travis which I don't understand. I openend a thread in their support forum, so let's see what happens.

> dyld: Library not loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libGIF.dylib
> 
>   Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
> 
>   Reason: Incompatible library version: ImageIO requires version 1.0.0 or later, but libGIF.dylib provides version 0.0.0

**Comment by Flamefire** [comment]
@Flow86 Can this be merged?

File: libs/libGamedata/lua/LuaInterfaceBase.cpp [hunk 1]
```diff
@@ -53,37 +53,32 @@ void LuaInterfaceBase::Register(kaguya::State& state)
 
 void LuaInterfaceBase::ErrorHandlerNoThrow(int /*status*/, const char* message)
 {
-    LOG.write(_("Lua error: %s\n")) % (message ? message : "Unknown");
+    LOG.write(_("Lua error: %s\n")) % (message ? message : _("Unknown"));
```
Comment (Flow86) [inline libs/libGamedata/lua/LuaInterfaceBase.cpp]: should'nt be this also "Unknown error"? to simplify translation?

File: libs/s25main/ingameWindows/iwOptionsWindow.cpp [hunk 2]
```diff
@@ -66,114 +85,83 @@ iwOptionsWindow::iwOptionsWindow()
...
-    AddTextButton(17, DrawPoint(67, 443), Extent(168, 24), TC_RED1, _("Surrender"), NormalFont);
-    // "Spiel beenden"
-    AddTextButton(0, DrawPoint(67, 474), Extent(168, 24), TC_RED1, _("End game"), NormalFont);
+    AddTextButton(ID_btMusicPlayer, DrawPoint(100, 380), Extent(160, 22), TC_GREEN2, _("Music player"), NormalFont);
+    AddTextButton(ID_btAdvanced, DrawPoint(67, 412), Extent(168, 24), TC_GREEN2, _("Advanced"), NormalFont);
+    AddTextButton(ID_btSurrender, DrawPoint(67, 443), Extent(168, 24), TC_RED1, _("Surrender"), NormalFont);
+    AddTextButton(ID_btEndGame, DrawPoint(67, 474), Extent(168, 24), TC_RED1, _("End game"), NormalFont);
 }
 
 void iwOptionsWindow::Msg_ButtonClick(const unsigned ctrl_id)
```
Comment (Flow86) [inline libs/s25main/ingameWindows/iwOptionsWindow.cpp]: we should perhaps some time migrate these multiplexing stuff to std::function/bind?

Comment (Flamefire) [reply libs/s25main/ingameWindows/iwOptionsWindow.cpp]: I was thinking about the same so you can do `AddTextButton(...).OnClick([this](){ this->doSomething(); }` and get rid of the IDs altogether. But this has downsides too and is a lot of work for not much gain

File: libs/libGamedata/lua/LuaInterfaceBase.cpp (hunk 1 above)
Comment (Flamefire) [reply libs/libGamedata/lua/LuaInterfaceBase.cpp]: Yes, done

File: libs/s25main/ingameWindows/iwOptionsWindow.cpp (hunk 2 above)
Comment (Flow86) [reply libs/s25main/ingameWindows/iwOptionsWindow.cpp]: exactly. The only downside (besides changing a lot of code) which comes in my mind is that you "pollute" your constructor instead? The whole (de)multiplexing would get removed by such an approach?

Comment (Flamefire) [reply libs/s25main/ingameWindows/iwOptionsWindow.cpp]: You'll also need complete types (But then we could get rid of the `AddFoo` functions) and you might want more than 1 line of code which doesn't look good in lambdas, hence you'll get many OnFooClicked functions or so.

**Pull request merged**